### ISSUE TITLE
Github.oauth_scopes property is None initially

### ISSUE THREAD

**Issue by dregad** [PyGithub/PyGithub#1943]
I do not understand why _oauth_scopes_ is not set when the object is created. It only gets populated after some method or property is called.

```python
from github import Github
token = 'xxx'  # Some Github token with at least 1 scope
gh = Github(token);
print(gh.oauth_scopes)  # None
gh.rate_limiting
print(gh.oauth_scopes)  # ['public_repo']
```

Python 3.8.5, PyGithub 1.55

**Comment by s-t-e-v-e-n-k** [comment]
Because we don't make a request until you call a property or a method.

**Comment by dregad** [comment]
Hi Steve, thanks for the fast reply. That makes sense. 

What confused me, was that when debugging, PyCharm is showing me the protected _requester.oauth_scopes having a value:

![image](https://user-images.githubusercontent.com/449891/117408672-8e7fa680-af10-11eb-947a-47bc421561b8.png)

I guess it is PyCharm's debugger itself that triggers a request when getting the *rate_limiting* property. DBP does not show this behavior until *gh.rate_limiting* is queried.

```
> /tmp/test.py(4)<module>()->None
-> breakpoint()
(Pdb) p gh.oauth_scopes
None
(Pdb) p gh._Github__requester.oauth_scopes
None
(Pdb) p gh.rate_limiting
(4985, 5000)
(Pdb) p gh.oauth_scopes
['admin:org', 'public_repo']
```

**Comment by mkenigs** [comment]
I ran into a bug because of this, because the docstring says it's a list of string: https://github.com/PyGithub/PyGithub/blob/84912a675cc09a57e947766674c462ddb0b45dbf/github/MainClass.py#L234

I think it either needs to be an empty list or the docstring should be updated.

**Comment by mkenigs** [comment]
It's also pretty confusing what will cause `oauth_scopes` to become something other than `None`, particularly when comparing between a token that has readonly access to a repo:
```
> cat reproducer.py
from github import Github

gh_with_write = Github("redacted")
gh_with_write.get_user()
print(gh_with_write.oauth_scopes)
gh_with_write.rate_limiting
print(gh_with_write.oauth_scopes)

gh_readonly = Github("redacted")
gh_readonly.get_user()
print(gh_readonly.oauth_scopes)
gh_readonly.rate_limiting
print(gh_readonly.oauth_scopes)
```

```
> python3 reproducer.py
None
['gist', 'read:org', 'repo', 'workflow']
None
None
```

**Comment by EnricoMi** [comment]
> I ran into a bug because of this, because the docstring says it's a list of string:
> 
> https://github.com/PyGithub/PyGithub/blob/84912a675cc09a57e947766674c462ddb0b45dbf/github/MainClass.py#L234
> 
> I think it either needs to be an empty list or the docstring should be updated.

You are right, the returned value is a string with comma-separated scopes. Would you mind creating a pull request to fix the doc string?

**Comment by EnricoMi** [comment]
> It's also pretty confusing what will cause `oauth_scopes` to become something other than `None`, particularly when comparing between a token that has readonly access to a repo

You are right, this is confusing and unexpected. Would you like to investigate this issue and sketch a fix?

Maybe accessing `gh.oauth_scopes` should trigger a simple API call to fetch the current oauth scopes and populate this field. Maybe there is even a specific API endpoint for that.
